osu.Game/Screens/Play/Player.cs

```diff
@@ -117,8 +117,12 @@ private void load(AudioManager audio, BeatmapDatabase beatmaps, OsuGameBase game
 
             ruleset = Ruleset.GetRuleset(usablePlayMode);
 
+            scoreProcessor = ruleset.CreateScoreProcessor(beatmap.HitObjects.Count);
+            scoreProcessor.Initialize(beatmap);

```

Comment:
Why is initialization and resetting not done inside the score processor constructor? In most cases `Initialize` methods are baaaaad, and an initial `Reset` should be automatic. In general it's a good principle to always make objects be in a valid and usable state after construction.